### PR TITLE
example: with-next-app

### PR DESCRIPTION
**Pull request by DanielSinclair**
- new example `examples/with-next-app`

### CODE REVIEW

File: examples/with-next-app/app/page.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+'use client';
```
Comment (jxom) [inline examples/with-next-app/app/page.tsx:1]: fwiw, i would avoid client components at the root and try to get it as close to the leaves as possible so conumers can still leverage server components if they want.

two options:
- compose `ConnectButton` into a separate component tagged as a client component,
- [add the `banner` option to esbuild config](https://github.com/wagmi-dev/wagmi/blob/main/packages/react/tsup.config.ts#L8-L10)

File: examples/with-next-app/app/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
+'use client';
```
Comment (jxom) [inline examples/with-next-app/app/layout.tsx:1]: also avoid `"use client";` here and encapsulate client dependants - good example of this is on create-wagmi: https://github.com/wagmi-dev/create-wagmi/tree/main/templates/next/rainbowkit/src/app

Comment (DanielSinclair) [reply examples/with-next-app/app/layout.tsx:1]: @jxom Great advice; have adopted the Wagmi `providers.tsx` approach.

File: examples/with-next-app/app/page.tsx (hunk 1 above)
Comment (DanielSinclair) [reply examples/with-next-app/app/page.tsx:1]: Ah, was wondering how to prevent esbuild from dropping that `use client;` tag. Thanks!

**Review by nickbytes — APPROVED** [review]

**Pull request merged**